certbot/certbot/compat/misc.py

```diff
@@ -116,23 +116,25 @@ def underscores_for_unsupported_characters_in_path(path):
     return drive + tail.replace(':', '_')
 
 
-def execute_command(cmd_name, shell_cmd):
-    # type: (str, str) -> Tuple[str, str]
+def execute_command(cmd_name, shell_cmd, env=None):
+    # type: (str, str, dict) -> Tuple[str, str]
```

Comment:
I think we'll also need to update the imports to import `Optional` from `acme.magic_typing`, but I think this should be:
```suggestion
    # type: (str, str, Optional[dict]) -> Tuple[str, str]
```

Reply:
I thought we would, but I looked at other examples in `compat` that have arguments with default values and apparently we don't do it there, so idk?

Reply:
Huh. I couldn't find anything quite as definitive as I would like, but it seems like it doesn't matter. See https://mypy.readthedocs.io/en/stable/command_line.html?highlight=implicit-optional#cmdoption-mypy-no-implicit-optional.

Reply:
oh, extremely neat. do you have a style preference then?

Reply:
Not really. Feel free to change it or leave it as is.